src/app/accounts/account-dialog/account-dialog.component.ts

```diff
@@ -34,16 +34,45 @@ export class AccountDialogComponent {
 
   createForm() {
     this.form = this.formBuilder.group({
-      account: "",
+      account: ["", this.uniqueNameValidator.bind(this)],
       balance: "",
       type: "",
       currency: ""
     });
+    this.addEventValidation([
+      this.form.controls.account,
+      this.form.controls.balance
+    ]);
   }
 
   submit(form: any) {
     const acc = Object.assign(form.value);
     acc.balance = Number.parseInt(acc.balance) || 0;
     this.matDialogRef.close(acc);
   }
+
+  addEventValidation(fields: AbstractControl[]) {
+    fields.forEach(field =>
+      field.valueChanges.subscribe(() => field.markAsTouched())
+    );
+  }
+
+  uniqueNameValidator(control: AbstractControl) {
+    const existing = !this.isCurrent(control.value)
+      ? this.exists(control.value)
+      : false;
+    return existing ? { existingName: { value: control.value } } : null;
+  }
+
+  isCurrent(name) {
+    return this.data.current
```

Comment:
this.data.current && (name.toLowerCase() === this.data.current.name.toLowerCase())